js/fetcher.js

```diff
@@ -0,0 +1,36 @@
+import { getFromLocalStorage } from "./common/utils/localStorageUtil.js";
+
+// Checks if token is present
+const accessToken = getFromLocalStorage("accessToken");
+console.log("Access Token:", accessToken);
+
+export async function fetcher(
+  url,
+  options = { method: "POST" },
+  shouldUseAuth = false
+) {
+  try {
+    let fetchOptions = {
+      ...options,
+      headers: { "Content-Type": "application/json" },
+    };
+
+    //if Auth, add token to header
+    if (shouldUseAuth) {
+      const accessToken = getFromLocalStorage("accessToken");
+      fetchOptions = {
+        ...fetchOptions,
+        headers: {
+          ...fetchOptions.headers,
+          Authorization: `Bearer ${accessToken}`,
+        },
+      };
+    }
+
+    const response = await fetch(url, fetchOptions);
+    const data = await response.json();
```

Comment:
I would check if the request was a success before returning the results/data. You can use the ok property to check, for example:

`if(response.ok) {
    return data;
 }
 // response not ok, throw an error, catch it in the try/catch of the calling function
 throw new Error(data.errors[0].message);
`